zulip/zulip/__init__.py

```diff
@@ -331,14 +357,14 @@ def __init__(self, email=None, api_key=None, config_file=None,
             if insecure is None and config.has_option("api", "insecure"):
                 # Be quite strict about what is accepted so that users don't
                 # disable security unintentionally.
-                insecure_setting = config.get("api", "insecure").lower()
-                if insecure_setting == "true":
-                    insecure = True
-                elif insecure_setting == "false":
-                    insecure = False
-                else:
-                    raise ZulipError("insecure is set to '%s', it must be 'true' or 'false' if it is used in %s"
-                                     % (insecure_setting, config_file))
+                insecure_setting = config.get('api', 'insecure')
+
+                insecure = validate_boolean_field(insecure_setting)
+
+                raise ZulipError("insecure is set to '{}', it must be 'true' "
+                                 "or 'false' if it is used in {}"
+                                 .format(insecure_setting, config_file))
```

Comment:
This seems wrong, since it's happening unconditionally...

Reply:
Whoops! I just pushed a fixed commit, thanks!